### DISCUSSION TITLE
`auth.updateUser()` allows users to set their email to the email of another account. How to prevent?

### DISCUSSION THREAD

**Discussion by Kilowhisky** [supabase/supabase#30344] [Questions]
Hi,

I'm trying to build a supabase based application where emails must be unique (OTP based login). 

The supabase-js lib allows users to `updateUser({ email })` to the email of another user and then when they confirm the change there are now two accounts with the same email address. 

I understand how to effectively do email change via API and 'do it right'. But I can't stop a user with a valid token from executing the `updateUser({ email })` on their own and corrupting the data. 

How do I prevent the user from doing this?  A trigger with a check and throw?

**Comment by GaryAustin1** [comment]
Are you speculating this or it is happening?  UpdateUser should error if the email exists and how would they confirm the email if they don't own it?
